### PR TITLE
ec2_snapshot - boto3 migration

### PR DESCRIPTION
**Pull request by tremble**
##### SUMMARY

Migrate ec2_snapshot to boto3

##### ISSUE TYPE

- Feature Pull Request

##### COMPONENT NAME

ec2_snapshot

##### ADDITIONAL INFORMATION

Assumes Python 3.x (datetime.timezone.utc)

### CODE REVIEW

File: plugins/modules/ec2_snapshot.py [hunk 1]
```diff
@@ -163,13 +164,14 @@ def _get_most_recent_snapshot(snapshots, max_snapshot_age_secs=None, now=None):
...
 
-    # See if the snapshot is younger that the given max age
-    snapshot_start = datetime.datetime.strptime(youngest_snapshot.start_time, '%Y-%m-%dT%H:%M:%S.%fZ')
-    snapshot_age = now - snapshot_start
+    youngest_snapshot = max(snapshots, key=lambda s: s['StartTime'])
+    snapshot_start = youngest_snapshot['StartTime']
+    try:
+        snapshot_age = now - snapshot_start
+    except Exception as e:
+        module.fail_json_aws(e, msg="bang", now=now, snapshot_start=snapshot_start)
```
Comment (jillr) [inline plugins/modules/ec2_snapshot.py:174]: Is "bang" just here for debugging?

**Review by jillr — COMMENTED** [review]
thanks @tremble, this looks pretty good.

File: plugins/modules/ec2_snapshot.py (hunk 1 above)
Comment (tremble) [reply plugins/modules/ec2_snapshot.py]: Yeah.  I was having fun with "offset aware/naïve" datetime objects.

If you see fail/exceptions with "bang"/boom" it's probably debugging that's snuck through.  Similarly catching Exception is usually me being lazy during debugging.

**Review by abikouo — APPROVED** [review]
LGTM

**Review by alinabuzachis — APPROVED** [review]

**Comment by tremble** [comment]
Two cloud LGTMs, letting zuul / gate do its thing

**Pull request merged**